### DISCUSSION TITLE
Default System Threads in Fprime

### DISCUSSION THREAD

**Discussion by satzguyz** [nasa/fprime#1407] [Q&A]
Hello,

What are all the list of default system threads usually initialized during the bootprocess ?

I assume all async handlers are invoked by default system thread. Is that understanding right ?

**Comment by timcanham** [comment]
The system thread calls all the `.start()` methods of the active components. The async handlers (input ports, commands) are done either on the thread of the active component or on the thread of a component calling a synchronous port. The the messages are then retrieved from the message queue by user code in the synchronous port handler. So, no, async handlers are not invoked by the default system thread.

**Comment by satzguyz** [comment]
This is very informative. Is there any order in which the start of various
components involved.

**Comment by timcanham** [comment]
Just the order of execution during initialization.
